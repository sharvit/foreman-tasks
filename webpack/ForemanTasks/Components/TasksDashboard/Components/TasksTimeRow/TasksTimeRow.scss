.tasks-time-row {
  margin: 0;
  padding: 0 10px;

  .time-label {
    font-size: 130%;
    font-weight: bold;
    margin-right: 10px;
    vertical-align: middle;
  }
}
