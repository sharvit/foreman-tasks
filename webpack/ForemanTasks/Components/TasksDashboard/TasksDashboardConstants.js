import { translate as __ } from 'foremanReact/common/I18n';

export const TASKS_DASHBOARD_AVAILABLE_QUERY_STATES = {
  RUNNING: 'running',
  PAUSED: 'paused',
  STOPPED: 'stopped',
  SCHEDULED: 'scheduled',
};

export const TASKS_DASHBOARD_AVAILABLE_QUERY_RESULTS = {
  ERROR: 'error',
  WARNING: 'warning',
  SUCCESS: 'success',
};

export const TASKS_DASHBOARD_AVAILABLE_QUERY_MODES = {
  LAST: 'last',
  OLDER: 'older',
};

export const TASKS_DASHBOARD_AVAILABLE_TIMES = {
  H24: 'H24',
  H12: 'H12',
  WEEK: 'WEEK',
};

export const TASKS_DASHBOARD_AVAILABLE_TIMES_TEXT = {
  [TASKS_DASHBOARD_AVAILABLE_TIMES.H24]: __('24h'),
  [TASKS_DASHBOARD_AVAILABLE_TIMES.H12]: __('12h'),
  [TASKS_DASHBOARD_AVAILABLE_TIMES.WEEK]: __('week'),
};

export const TASKS_DASHBOARD_CURRENT_TIME = 'TASKS_DASHBOARD_CURRENT_TIME';

export const FOREMAN_TASKS_DASHBOARD_INIT = 'FOREMAN_TASKS_DASHBOARD_INIT';

export const FOREMAN_TASKS_DASHBOARD_UPDATE_TIME =
  'FOREMAN_TASKS_DASHBOARD_UPDATE_TIME';

export const FOREMAN_TASKS_DASHBOARD_UPDATE_QUERY =
  'FOREMAN_TASKS_DASHBOARD_UPDATE_QUERY';

export const MOCKED_DATA = {
  running: {
    last: 3,
    older: 5,
  },
  paused: {
    last: 3,
    older: 5,
  },
  stopped: {
    error: {
      total: 8,
      last: 1,
    },
    warning: {
      total: 20,
      last: 2,
    },
    success: {
      total: 25,
      last: 3,
    },
  },
  scheduled: 1,
};
