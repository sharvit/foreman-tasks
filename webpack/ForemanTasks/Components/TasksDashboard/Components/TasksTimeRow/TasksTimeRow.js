import React from 'react';
import PropTypes from 'prop-types';
import { Row } from 'patternfly-react';
import { noop } from 'foremanReact/common/helpers';
import { translate as __ } from 'foremanReact/common/I18n';

import { timePropType } from '../../TasksDashboardPropTypes';
import { TASKS_DASHBOARD_AVAILABLE_TIMES } from '../../TasksDashboardConstants';
import TimeDropDown from './Components/TimeDropDown/TimeDropDown';
import './TasksTimeRow.scss';

const TasksTimeRow = ({ time, updateTime }) => (
  <Row className="tasks-time-row">
    <span className="time-label">{__('With focus on last')}</span>
    <TimeDropDown
      id="tasks-dashboard-time-period-dropdown"
      selectedTime={time}
      onChange={updateTime}
    />
  </Row>
);

TasksTimeRow.propTypes = {
  time: timePropType,
  updateTime: PropTypes.func,
};

TasksTimeRow.defaultProps = {
  time: TASKS_DASHBOARD_AVAILABLE_TIMES.H24,
  updateTime: noop,
};

export default TasksTimeRow;
