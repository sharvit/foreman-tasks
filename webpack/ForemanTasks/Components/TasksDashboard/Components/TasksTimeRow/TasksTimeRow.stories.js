import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { withCardsDecorator } from '../../../../../stories/decorators';

import { TASKS_DASHBOARD_AVAILABLE_TIMES } from '../../TasksDashboardConstants';
import TasksTimeRow from './TasksTimeRow';

storiesOf('TasksDashboard/TasksTimeRow', module)
  .addDecorator(withKnobs)
  .addDecorator(withCardsDecorator)
  .add('TasksTimeRow', () => (
    <TasksTimeRow
      time={select(
        'time',
        TASKS_DASHBOARD_AVAILABLE_TIMES,
        TASKS_DASHBOARD_AVAILABLE_TIMES.H24
      )}
      updateTime={action('updateTime')}
    />
  ));
