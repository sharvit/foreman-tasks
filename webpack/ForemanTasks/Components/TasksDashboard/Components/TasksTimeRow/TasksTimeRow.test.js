import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import { TASKS_DASHBOARD_AVAILABLE_TIMES } from '../../TasksDashboardConstants';
import TasksTimeRow from './TasksTimeRow';

const fixtures = {
  'render with minimal props': {},
  'render with props': {
    time: TASKS_DASHBOARD_AVAILABLE_TIMES.WEEK,
    updateTime: jest.fn(),
  },
};

describe('TasksTimeRow', () =>
  testComponentSnapshotsWithFixtures(TasksTimeRow, fixtures));
