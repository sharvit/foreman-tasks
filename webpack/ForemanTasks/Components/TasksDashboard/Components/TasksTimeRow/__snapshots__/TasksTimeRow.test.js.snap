// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksTimeRow render with minimal props 1`] = `
<Row
  bsClass="row"
  className="tasks-time-row"
  componentClass="div"
>
  <span
    className="time-label"
  >
    With focus on last
  </span>
  <TimeDropDown
    className=""
    id="tasks-dashboard-time-period-dropdown"
    onChange={[Function]}
    selectedTime="H24"
  />
</Row>
`;

exports[`TasksTimeRow render with props 1`] = `
<Row
  bsClass="row"
  className="tasks-time-row"
  componentClass="div"
>
  <span
    className="time-label"
  >
    With focus on last
  </span>
  <TimeDropDown
    className=""
    id="tasks-dashboard-time-period-dropdown"
    onChange={[MockFunction]}
    selectedTime="WEEK"
  />
</Row>
`;
